### PR TITLE
fix(portable-text): shallow-depth assumptions in the Portable Text Input

### PR DESCRIPTION
**Pull request by christianhg**
### Description

With #13315 Studio's PT UI now composes nicely with PTE Containers. However, shallow-depth assumptions in Studio still broke certain functionality.

#### The four fixes

**`fix(portable-text): resolve render-callback schema types by path`** — `Compositor`'s `renderBlockObject`, `renderInlineObject`, and `renderAnnotation` looked schema types up by scanning the flat `schemaTypes.{blockObjects,inlineObjects,annotations}` arrays. Those arrays only describe the root portable-text array, so container-nested types would either miss-and-throw or get the wrong schema type. The fix is to resolve each type with `getSanitySubSchema` from `@portabletext/sanity-bridge` which fetches the correct sub schema depending on the level of nesting inside PTE.

**`fix(portable-text): collect form members at any depth`** — `usePortableTextMemberItemsFromProps` walked the root `members` array flat and didn't look inside block object fields for nested arrays. That left Container descendants missing from the precomputed `portableTextMemberItems` list, so the focus-tracking and member-input lookups that consult it never found them. The fix is to recursively descend into "array of objects" field members.

**`fix(portable-text): track focus into nested-container blocks`** — `useTrackFocusPath` mapped the form focusPath to an editor selection by reading absolute path segments (`focusPath.length === 1` for a block, `focusPath[1] === 'children'` for a span, `focusPath[3] === 'text'` for span text). Those shape checks only hold at the root of the Portable Text document, so a `focusPath` into a nested-container block fell through every branch silently. The fix is to compose first-class PTE traversal utils to verify what the path is pointing to.

**`fix(portable-text): key the element-ref registry by full path`** — the registry that maps members to their rendered DOM elements used `memberItem.member.key` (the member's bare local `_key`). Keys in PTE are unique inside their parent array, not across the document, so two spans keyed `s0` (one in each block) overwrote each other's element refs. The fix is to key by path instead of `_key`.

### What to review

I deliberately tried to keep this PR small and focused. Primary changes are in `usePortableTextMemberItemsFromProps` and `useTrackFocusPath`. The rest is simple call-site updates.

Note: This PR fixes logic related to focus paths pointing at spans, but that actually doesn't seem to work since the form store's members list doesn't include spans. Maybe something to look into in a separate PR.

### Testing

1. Open a document in https://test-studio-git-feat-pt-path-aware-lookups.sanity.dev/test/structure/input-standard;portable-text;customPlugins and add a table in the "Container Table" PT Input.
2. Add a image or an inline note in a table cell and verify that you can double-click to edit.
3. Add an annotation to some text in a table cell.
4. To test focus tracking, the "easiest" way is to construct a URL like `https://test-studio-git-feat-pt-path-aware-lookups.sanity.dev/test/structure/input-standard;portable-text;customPlugins;717a1bff-0a8d-4cad-b531-46425e74ef4c,path=containerTable[_key=="445810a7bb73"].rows[_key=="2e4bbfb36491"].cells[_key=="439c24e0a1e4"].content[_key=="be1d49ff4c72"]` and verify that Studio puts the caret in the text block nested in the table cell.
5. The "key the element-ref registry by full path" fix is more a correctness fix that is hard to test since it requires objects with duplicate keys in the document.

### Notes for release

n/a

---

> [!NOTE]
> **Medium Risk**
> Changes core portable-text focus, schema resolution, and member enumeration used by editing modals and deep-link URLs; behavior is localized to PT input with new browser coverage but regressions could affect nested-container editing and scroll/focus sync.
> 
> **Overview**
> Fixes **Portable Text Input** behavior when content lives inside PTE **containers** (e.g. table cells), where the UI previously assumed root-level paths and flat schema/member lists.
> 
> **`Compositor`** resolves block objects, inline objects, and annotations with **`getSanitySubSchema`** at each node’s editor path (plus editor snapshot) instead of scanning root-only `schemaTypes.*` arrays, so nested types render with the correct Sanity schema.
> 
> **`usePortableTextMemberItemsFromProps`** iteratively walks into **array-of-objects** fields inside object blocks so annotations, inline/block objects, and nested text blocks are included in the flat member list used for inputs and focus tracking. DOM element refs are keyed by **`memberItem.key`** (full path string), aligning lookups with **`usePortableTextMemberItem(pathToString(...))`** and avoiding `_key` collisions across depths.
> 
> **`useTrackFocusPath`** takes **`ptInputPath`**, uses **`getEnclosingBlock`** / **`getSpan`** to map form `focusPath` to editor-relative selection, and matches members by **absolute document path** rather than shallow segment shapes (`focusPath.length === 1`, etc.).
> 
> Browser tests add a **nested table** fixture and cover focus tracking for spans and blocks inside containers.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 32ada349a92bbce55189a347ca260b266c37073f. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/sanity/src/core/form/inputs/PortableText/hooks/useTrackFocusPath.tsx [hunk 1]
```diff
@@ -95,48 +98,31 @@ export function useTrackFocusPath(props: Props): void {
...
-
-        // Is focus directly on a text block child?
-        const isTextChildFocusPath =
-          isTextBlock &&
-          ((focusPath.length === 3 && focusPath[1] === 'children') || isSpanTextFocusPath)
+      if (isTextBlock || relatedEditorItem.kind === 'objectBlock') {
+        // Resolve block and span from the snapshot — in nested containers
+        // the block path can be longer than one segment.
+        const snapshot = editor.getSnapshot()
+        const enclosingBlock = getEnclosingBlock(snapshot, focusPath)
```
[the remark this answers is not in the document]
Comment (christianhg) [reply packages/sanity/src/core/form/inputs/PortableText/hooks/useTrackFocusPath.tsx]: Done!

File: packages/sanity/src/core/form/inputs/PortableText/hooks/usePortableTextMembers.tsx [hunk 2]
```diff
@@ -61,11 +61,24 @@ export function usePortableTextMemberItemsFromProps(
...
         const isObjectBlock = !isBlockType(member.item.schemaType)
         if (isObjectBlock) {
           result.push({kind: 'objectBlock', member, node: member.item})
+          // Non-PT array fields surface no block-shaped items and
+          // contribute nothing, so pushing them is safe.
+          for (const field of member.item.members) {
+            if (isArrayOfObjectsFieldMember(field)) {
+              stack.push(field.field.members)
+            }
+          }
```
[the remark this answers is not in the document]
Comment (christianhg) [reply packages/sanity/src/core/form/inputs/PortableText/hooks/usePortableTextMembers.tsx]: Done!

File: packages/sanity/src/core/form/inputs/PortableText/hooks/useTrackFocusPath.tsx [hunk 3]
```diff
@@ -23,7 +25,8 @@ export function useTrackFocusPath(props: Props): void {
 
   const portableTextMemberItems = usePortableTextMemberItems()
   const elementRefs = usePortableTextMemberItemElementRefs()
-  const editor = usePortableTextEditor()
+  const editor = useEditor()
+  const legacyEditor = usePortableTextEditor()
   const selection = usePortableTextEditorSelection()
```
[the remark this answers is not in the document]
Comment (christianhg) [reply packages/sanity/src/core/form/inputs/PortableText/hooks/useTrackFocusPath.tsx:31]: Feels unrelated to this PR to refactor that.

File: packages/sanity/src/core/form/inputs/PortableText/hooks/useTrackFocusPath.tsx [hunk 4]
```diff
@@ -73,9 +76,27 @@ export function useTrackFocusPath(props: Props): void {
...
-    const relatedEditorItem = focusedItem || openItem
-    const elementRef = relatedEditorItem ? elementRefs[relatedEditorItem.member.key] : undefined
+    // Resolve the enclosing block for focusPath from the editor snapshot.
+    const snapshot = editor.getSnapshot()
+    const enclosingBlock = getEnclosingBlock(snapshot, focusPath)
+    const blockPath = enclosingBlock?.path
+    const span = getSpan(snapshot, focusPath)
+
+    // `member.item.path` is doc-absolute; `blockPath` is input-relative, so compare tails.
+    const enclosingBlockItem = blockPath
```
Comment (jordanl17) [inline packages/sanity/src/core/form/inputs/PortableText/hooks/useTrackFocusPath.tsx:89]: This might be nothing, but flagging just in case. The tail comparison for matching the enclosing block made me pause: if two blocks happen to share the same `_key` at different depths (which I think is valid, since keys are only unique within their parent array?), wouldn’t the tail match hit both and .find just take whichever comes first? Wondered if matching on the full path might be safer?

Comment (christianhg) [reply packages/sanity/src/core/form/inputs/PortableText/hooks/useTrackFocusPath.tsx:89]: Nice catch! I think you are right, this is unsafe. Will switch to full path comparison.

Comment (christianhg) [reply packages/sanity/src/core/form/inputs/PortableText/hooks/useTrackFocusPath.tsx:89]: Fixed! Thank you for flaggin!

**Review by jordanl17 — APPROVED** [review]
Great stuff! Thanks @christianhg
